src/fsharp/vs/Exprs.fs

```diff
@@ -752,6 +679,190 @@ module FSharpExprConvert =
+                        failwith "Failed to recognize record type member"
+                elif enclosingEntity.IsUnionTycon then
+                    if vName = "GetTag" then
+                        let objR = ConvExpr cenv env callArgs.Head
+                        E.UnionCaseTag(objR, typR) 
+                    elif vName.StartsWith("New") then
+                        let name = vName.Substring(3)
+                        let mkR = ConvUnionCaseRef cenv (UCRef(tcref, name))
+                        let argsR = ConvExprs cenv env callArgs
+                        E.NewUnionCase(typR, mkR, argsR)
+                    elif vName.StartsWith("Is") then
+                        let name = vName.Substring(2)
+                        let mkR = ConvUnionCaseRef cenv (UCRef(tcref, name))
+                        let objR = ConvExpr cenv env callArgs.Head
+                        E.UnionCaseTest(objR, typR, mkR)
+                    else 
+                        match subClass with
+                        | Some name ->
+                            let ucref = UCRef(tcref, name)
+                            let mkR = ConvUnionCaseRef cenv ucref                                        
+                            let objR = ConvLValueExpr cenv env callArgs.Head
+                            let projR = FSharpField(cenv, ucref, ucref.Index)
+                            E.UnionCaseGet(objR, typR, mkR, projR)
+                        | _ ->
+                            failwith "Failed to recognize union type member"
+                else
+                    let names = enclosingEntity.MembersOfFSharpTyconSorted |> List.map (fun v -> v.CompiledName) |> String.concat ", "
+                    failwithf "Member '%s' not found in type %s, found: %s" vName enclosingEntity.DisplayName names
+            | _ -> // member is overloaded
+                match nlr.ItemKey.TypeForLinkage with
+                | None -> failwith "Type of signature could not be resolved"
+                | Some keyTy ->
+                    let findBySig =
+                        findByName |> List.tryFind (fun v -> ccu.MemberSignatureEquality(keyTy,v.Type))
+                    match findBySig with
+                    | Some v ->
+                        makeFSCall isMember v
+                    | _ ->
+                        failwith "Failed to recognize F# member"
+
+        // First try to resolve it to IL metadata
+        let try1 = 
+            if tcref.IsILTycon then 
+                try 
+                    let mdef = resolveILMethodRefWithRescope unscopeILType tcref.ILTyconRawMetadata ilMethRef 
+                    let minfo = MethInfo.CreateILMeth(cenv.amap, m, enclosingType, mdef)                     
+                    FSharpMemberOrFunctionOrValue(cenv, minfo) |> makeCall |> Some
+                with _ -> 
+                    None
+            else
+                None
+
+        // Otherwise try to bind it to an F# symbol
+        match try1 with
```

Comment:
Could you extract this large expression out to a separate helper function please, outside the big ``let rec``? Along the lines of ``bindILMethodRefToSymbol`` in the previous code. Thanks